src/courses/mappers/14.md

```diff
@@ -0,0 +1,1697 @@
+
+::: details JSON Mapper Test Skeleton
+```typescript
+import fs from 'fs';
+import {SKELETONMapper} from '../../../src/SKELETON-mapper';
+import {omitVersions} from '../../utils';
+
+describe('SKELETON_mapper', () => {
+  it('Successfully converts SKELETON targeted at a local/cloned repository data', () => {
+    const mapper = new SKELETONMapper(
+      fs.readFileSync(
+        'sample_jsons/SKELETON_mapper/sample_input_report/SKELETON.json',
+        {encoding: 'utf-8'}
+      )
+    );
+
+    // fs.writeFileSync(
+    //   'sample_jsons/SKELETON_mapper/SKELETON-hdf.json',
+    //   JSON.stringify(mapper.toHdf(), null, 2)
+    // );
+
+    expect(omitVersions(mapper.toHdf())).toEqual(
+      omitVersions(
+        JSON.parse(
+          fs.readFileSync(
+            'sample_jsons/SKELETON_mapper/SKELETON-hdf.json',
+            {
+              encoding: 'utf-8'
+            }
+          )
+        )
+      )
+    );
+  });
+});
+
+describe('SKELETON_mapper_withraw', () => {
+  it('Successfully converts withraw flagged SKELETON targeted at a local/cloned repository data', () => {
+    const mapper = new SKELETONMapper(
+      fs.readFileSync(
+        'sample_jsons/SKELETON_mapper/sample_input_report/SKELETON.json',
+        {encoding: 'utf-8'}
+      ),
+      true
+    );
+
+    // fs.writeFileSync(
+    //   'sample_jsons/SKELETON_mapper/SKELETON-hdf-withraw.json',
+    //   JSON.stringify(mapper.toHdf(), null, 2)
+    // );
+
+    expect(omitVersions(mapper.toHdf())).toEqual(
+      omitVersions(
+        JSON.parse(
+          fs.readFileSync(
+            'sample_jsons/SKELETON_mapper/SKELETON-hdf-withraw.json',
+            {
+              encoding: 'utf-8'
+            }
+          )
+        )
+      )
+    );
+  });
+});
+```
+:::
+
+### Fingerprinting
+
+OHDF Converters has a fingerprinting service that detects a security tool data format and automatically applies the correct mapper to convert it to OHDF. To enable this feature, we need to explicitly declare keywords unique to the security tool data format as follows:
+
+1. Go to the file `report_intake.ts` under the `heimdall2/apps/frontend/src/store` directory.
+
+2. Import your mapper file. You should be able to add the name of your mapper class to a pre-existing import statement pointing at `hdf-converters` as follows:
+```typescript
+import {
+  ASFFResults as ASFFResultsMapper,
+  BurpSuiteMapper,
+  ...
+  {YOUR-MAPPER-CLASS-HERE}
+} from '@mitre/hdf-converters';
+```
+
+3. Instantiate your mapper class in the `convertToHdf` switch block. Add the following lines:
+```typescript
+case '{YOUR-EXPORT-SERVICE-NAME-HERE}':
+  return new {YOUR-MAPPER-CLASS-HERE}(convertOptions.data).toHdf();
+```
+
+4. Navigate to the file `fingerprinting.ts` in the `src/utils` directory in `hdf-converters`. Add keywords that are unique to your sample export to the `fileTypeFingerprints` variable. It should be formatted as follows:
```

Comment:
missing the step to add the new mapper to the hdf converters readme and to the supported formats blurb on the heimdall ui 
![image](https://github.com/mitre/saf-training/assets/2015821/9062e700-85c1-40a1-b5b8-07caca50f003)